plugins/obs-webrtc/whip-output.cpp

```diff
@@ -401,7 +401,15 @@ bool WHIPOutput::Connect()
 	response.erase(0, response.find("v=0"));
 
 	rtc::Description answer(response, "answer");
-	peer_connection->setRemoteDescription(answer);
+	try {
+		peer_connection->setRemoteDescription(answer);
+	} catch (const std::exception &err) {
+		do_log(LOG_ERROR, "WHIP server responded with invalid SDP: %s",
```

Comment:
Can we catch a specific error class here for invalid SDP?

Reply:
I looked a little bit, I found a three different errors thrown
`logic_error` and `invalid_argument` and `runtime_error`. 

What would you do differently depending on the error type?

I am concerned that different error types could be thrown in the future (or old ones removed)

Reply:
I think the author of libdatachannel generally uses `std::invalid_argument` for invalid inputs, and that is unlikely to change in the future (why would they want to throw a spanner in the works for all their clients :smile:) - therefore in this case we can use that for invalid SDP.

In all other cases (including additional error types introduced in the future), we want to avoid a crash, but it is evidently a condition we have not thought about or did not expect, so we should leave a graceful error message, fail connecting, and wait on a bug report. In that case handle `std::exception`.

I've done that and uploaded a new patch.